### DISCUSSION TITLE
install x64 architecture

### DISCUSSION THREAD

**Discussion by azinsharaf** [microsoft/winget-cli#3321] [Q&A] [answered]
hi, how can i install an app with x64 architecture? i use this command but doesn't recognize the `-a` arg value.

`winget install -e --id Microsoft.DotNet.DesktopRuntime.6 –-architecture amd64 --accept-package-agreements –-silent`
or 
`winget install -e --id Microsoft.DotNet.DesktopRuntime.6 –-architecture x64 --accept-package-agreements –-silent`

error:
`Found a positional argument when none was expected: 'amd64'`

**Comment by denelon** [comment]
The enumerations of architectures currently supported is:
https://github.com/microsoft/winget-cli/blob/83cd0dcaddd31cc64570422b081f7743a79dc10d/schemas/JSON/manifests/v1.5.0/manifest.installer.1.5.0.json#L112-L119

There might be a bug in here with the argument parsing for the "x64" example.
@JohnMcPMS to confirm

**Comment by JohnMcPMS** [comment, marked as answer]
The problem (as pasted in the amd64 example) is that you have the sneaky U+2013 "En Dash" character as the first character in your `--architecture`.  There is certainly an opportunity for us to normalize and support these kinds of "alternate hyphens" that often use exactly the same glyph in common monospace console fonts, but we do not right now.

If you are using PowerShell, it can indicate the presence of these imposter hyphens.  It will not make the argument name dark grey if one of the first two characters is one of these dashes.

As an additional note, we only support `x64`, not `amd64` as input (also a place where we could make things more flexible).

**Reply by azinsharaf** [reply]
wow good catch! It works now. The `--silent` had the same issue. 
![image](https://github.com/microsoft/winget-cli/assets/10101468/7b1facb6-93ad-467c-8411-1e07a5fe8606)
